includes/class-sensei-utils.php

```diff
@@ -2626,6 +2626,51 @@ public static function format_last_activity_date( string $date ) {
 
 		return wp_date( get_option( 'date_format' ), $date->getTimestamp(), $timezone );
 	}
+
+	/**
+	 * Get's the HTML content from the Featured Video for a post
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @param string $post_id the post ID.
+	 *
+	 * @return string|false The featured video HTML output if it exists, or false if it doesn't.
+	 */
+	public static function get_featured_video_html( $post_id ) {
+		$video_embed  = '';
+		$allowed_html = array(
+			'embed'  => array(),
+			'iframe' => array(
+				'title'           => array(),
+				'width'           => array(),
+				'height'          => array(),
+				'src'             => array(),
+				'frameborder'     => array(),
+				'allowfullscreen' => array(),
+			),
+			'video'  => Sensei_Wp_Kses::get_video_html_tag_allowed_attributes(),
+		);
+		if ( has_blocks( $post_id ) ) {
+			$post   = get_post( $post_id );
+			$blocks = parse_blocks( $post->post_content );
+			foreach ( $blocks as $block ) {
+				if ( 'sensei-lms/featured-video' === $block['blockName'] ) {
+					if ( 'sensei-pro/interactive-video' === $block['innerBlocks'][0]['blockName'] ) {
+						$block = $block['innerBlocks'][0];
+					}
+					$video_embed = $block['innerBlocks'][0]['attrs']['url'];
+				}
+			}
+		} else {
+			$video_embed = get_post_meta( $post_id, '_lesson_video_embed', true );
+		}
+		if ( 'http' === substr( $video_embed, 0, 4 ) ) {
```

Comment:
We should avoiding duplicating code like this. Instead, let's create a new method, for example `render_video_embed`, that can be used in both methods.

Reply:
Done! ✅